Makefile

```diff
@@ -537,11 +547,14 @@ release-windows: release-windows-unsigned
 # details.
 #
 .PHONY: release-connect
-release-connect:
+release-connect: | $(RELEASE_DIR)
 	$(eval export CSC_NAME)
 	yarn install --frozen-lockfile
 	yarn build-term
 	yarn package-term -c.extraMetadata.version=$(VERSION)
+	if [ -n "$$CONNECT_TSH_APP_PATH" ]; then \
```

Comment:
Do we need to check `CONNECT_TSH_APP_PATH` before copying the dmg? The comment above `release-connect` says that this target is used only in macOS, in which case the dmg should always be present after a successful `yarn package-term`.

Reply:
This is discriminating between `CONNECT_TSH_APP_PATH` and `CONNECT_TSH_BIN_PATH`. The former is set when we are building release packages. On a push build we don't care about saving the package, and that uses `CONNECT_TSH_BIN_PATH`.

https://github.com/gravitational/teleport.e/blob/branch/v12/.github/workflows/build-mac-amd64.yaml#L193 and the step after the highlighted line.

I will shortly be revisiting the whole copying artifact thing when we stop using Drone to do any of this. I may revise this logic then to be more straightforward.

Reply:
Oops, I forgot that it does push builds too, I thought it's just for tag builds.